normandy/recipes/management/commands/update_addon_urls.py

```diff
@@ -0,0 +1,52 @@
+import json
+
+from urllib.parse import urlparse, urlunparse
+from django.core.management.base import BaseCommand
+
+from normandy.recipes.models import RecipeRevision
+
+
+class Command(BaseCommand):
+    """
+    Rewrite all revisions to update the URL of add-ons stored in Normandy's
+    file storage to a new hostname.
+    """
+
+    help = "Updates add-on URL in revisions"
+
+    def add_arguments(self, parser):
+        parser.add_argument("new_hostname", help="Hostname to convert add-on urls to")
+
+    def handle(self, *args, new_hostname, **options):
+        target_revisions = RecipeRevision.objects.filter(action__name="opt-out-study")
+        update_count = 0
+        for rev in target_revisions:
```

Comment:
Are there many of these in the database where this matters? Would it be worth changing this line to `for rev in target_revisions.iterator():` which will do the same thing but only make iterated lists of 2,000 instances at a time.

Reply:
I've never seen `.iterator()` before, that's interesting. I don't think it will matter but it is semantically more what we want here.

I checked [the api](https://normandy.cdn.mozilla.net/api/v3/recipe_revision/) and there are currently 1989 revisions. So if `.iterator()` makes chunks of 2000, then it won't have any effect.